### PR TITLE
Mobile: Add documentation for 207 and 502 responses

### PR DESCRIPTION
**Pull request by Tonksthebear**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

Add documentation for 207 and 502 responses

## Related issue(s)
- [department-of-veterans-affairs/va.gov-team#4643](https://app.zenhub.com/workspaces/va-mobile---backend-61d75d03a7505500147ec7f6/issues/gh/department-of-veterans-affairs/va-mobile-app/4643)
- https://github.com/department-of-veterans-affairs/va-mobile-app/issues/4643

## Testing done

None, just a documentation change

## What areas of the site does it impact?
Mobile Documentation

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/mobile/docs/schemas/ClaimsAndAppealsOverviewAppealsError.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  service:
+    type: string
+    enum:
+      - appeals
+      - claims
+  errorDetails:
+    type: array
+    items:
+      type: array
+      description: Array of arrays from failing upstream services. Used for debugging only.
```
Comment (kpethtel) [inline modules/mobile/docs/schemas/ClaimsAndAppealsOverviewAppealsError.yml:15]: I'm not sure if this is accurate. Looking at an example from the specs, I think errorDetails is an array of objects, not an array of arrays. I think you may have an extra layer of depth in the docs than exists in the response. 
```
"meta"=>
  {"errors"=>
    [{"service"=>"appeals",
      "errorDetails"=>
       [{"title"=>"Bad Gateway",
         "detail"=>"Received a 500 response from the upstream server",
         "code"=>"CASEFLOWSTATUS500",
         "source"=>"No method error",
         "status"=>"502"}]}],
```

**Review by kpethtel — COMMENTED** [review]
I'm not sure what the right approach to documenting this is, but I think the way it's written here, you're effectively only capturing the `error` part of the response. The schema you're replacing within the 207 response section is an example of a full response with an error section. This is just the error section with no context. I'm not sure if there's a clean and DRY way of using `modules/mobile/docs/schemas/ClaimsAndAppealsOverview.yml` but replacing the error section.

File: modules/mobile/docs/schemas/ClaimsAndAppealsOverviewAppealsError.yml (hunk 1 above)
Comment (Tonksthebear) [reply modules/mobile/docs/schemas/ClaimsAndAppealsOverviewAppealsError.yml]: Oo thanks for catching that. I just edited the original documentation which had the double nested arrays but I should have noticed that difference

**Comment by Tonksthebear** [comment]
Ya I spent a decent amount of time trying to make it DRY but I couldn't find a clean way of doing it. I added the successful service details in the 207 though

File: modules/mobile/docs/schemas/ClaimsAndAppealsOverviewAppealsError.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            - service
+            - errorDetails
+          properties:
+            service:
+              type: string
+              example: "appeals"
+              enum:
+                - appeals
+            errorDetails:
+              type: object
```
Comment (kpethtel) [inline modules/mobile/docs/schemas/ClaimsAndAppealsOverviewAppealsError.yml:64]: Looking at the example I pulled from the specs, it looks like errorDetails is an array of objects.

**Review by kpethtel — CHANGES_REQUESTED** [review]
It looks like the meta should also be an object. It's very likely that's true for all of our endpoints and the docs may be wrong for all of them. I get the impression a lot of these schemas are probably structured incorrectly. It definitely makes it harder to update them when they're wrong to begin with.

LMK if I'm reading this wrong.

File: modules/mobile/docs/schemas/ClaimsAndAppealsOverviewAppealsError.yml (hunk 2 above)
Comment (Tonksthebear) [reply modules/mobile/docs/schemas/ClaimsAndAppealsOverviewAppealsError.yml]: Ah you're right, thanks for catching that

**Comment by Tonksthebear** [comment]
Ok I think I cause all of the schema differences. You've been very helpful with looking this over, I only compared to other docs, not the schemas. You're probably right in that the docs are probably wrong for other schemas as well...

**Review by kpethtel — APPROVED** [review]
The "both" file doesn't go into the same level of detail about the error bodies as the individual error schemas do, but if someone wants that detail they can look at the individual errors. 👍

**Pull request merged**